ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/tokenize/SparseTokenizerModel.java

```diff
@@ -0,0 +1,136 @@
+        ModelResultFilter resultFilter = textDocsInput.getResultFilter();
+        for (String doc : textDocsInput.getDocs()) {
+            Output output = new Output(200, "OK");
+            Encoding encodings = tokenizer.encode(doc);
+            long[] indices = encodings.getIds();
+            List<ModelTensor> outputs = new ArrayList<>();
+            String[] tokens = Arrays.stream(indices)
+                    .distinct()
+                    .mapToObj(value -> new long[]{value})
+                    .map(value -> this.tokenizer.decode(value, true))
+                    .filter(s -> !s.isEmpty())
+                    .toArray(String[]::new);
+            Map<String, Float> tokenWeights = Arrays.stream(tokens)
+                    .collect(Collectors.toMap(
+                            token -> token,
+                            token -> idf.getOrDefault(token, 1.0f)
+                    ));
+            Map<String, ?> wrappedMap = Map.of(ML_MAP_RESPONSE_KEY, Collections.singletonList(tokenWeights));
+            ModelTensor tensor = ModelTensor.builder()
+                    .dataAsMap(wrappedMap)
+                    .build();
+            outputs.add(tensor);
+            ModelTensors modelTensorOutput = new ModelTensors(outputs);
+            output.add(modelTensorOutput.toBytes());
+            tensorOutputs.add(parseModelTensorOutput(output, resultFilter));
+        }
+        return new ModelTensorOutput(tensorOutputs);
+    }
+
+    protected void doLoadModel(List<Predictor<Input, Output>> predictorList, List<ZooModel<Input, Output>> modelList,
+                               String engine,
+                               Path modelPath,
+                               MLModelConfig modelConfig) throws ModelNotFoundException, MalformedModelException, IOException, TranslateException {
+        tokenizer = HuggingFaceTokenizer.builder().optPadding(true).optTokenizerPath(modelPath.resolve("tokenizer.json")).build();
+        idf = new HashMap<>();
+        if (Files.exists(modelPath.resolve("idf.json"))){
+            Gson gson = new Gson();
+            Type mapType = new TypeToken<Map<String, Float>>() {}.getType();
+            idf = gson.fromJson(new InputStreamReader(Files.newInputStream(modelPath.resolve("idf.json"))), mapType);
+        }
+        log.info("tokenize Model {} is successfully deployed", modelId);
+    }
+    @Override
+    public boolean isModelReady() {
+        if (modelHelper == null || modelId == null || tokenizer == null) {
+            return false;
+        }
+        return true;
+    }
+
+    @Override
+    public void close() {
+        if (modelHelper != null && modelId != null) {
+            modelHelper.deleteFileCache(modelId);
+            if (predictors != null) {
+                closePredictors(predictors);
+                predictors = null;
+            }
+            if (models != null) {
+                closeModels(models);
+                models = null;
+            }
```

Comment:
This part seems can be used for all cases(dense, sparse, token), check if this can be extracted to parent class.

Reply:
Already remove these part in tokenize model because tokenize model doesn't have model or predictors. For sparse and dense, they are already in textEmbedding model.